File: solutions/_118_pascals_triangle/solution.java
package _118_pascals_triangle;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {

  public List<List<Integer>> generate(int numRows) {
    List<List<Integer>> triangle = new ArrayList<>();
    if (numRows < 1) {
      return triangle;
    }

    triangle.add(Arrays.asList(1));
    for (int i = 1; i < numRows; i++) {
      List<Integer> row = new ArrayList<>();
      row.add(1);
      for (int j = 1; j < i; j++) {
        row.add(triangle.get(i - 1).get(j - 1) + triangle.get(i - 1).get(j));
      }
      row.add(1);
      triangle.add(row);
    }
    return triangle;
  }
}
/*
https://leetcode.com/problems/pascals-triangle/
*/

File: solutions/_118_pascals_triangle/solution.py
from typing import List

class Solution:
    def generate(self, numRows) -> List[List[int]]:
        if numRows < 1:
            return []

        triangle = [[1]]
        for i in range(1, numRows):
            row = [1]
            for j in range(1, i):
                row.append(triangle[i - 1][j - 1] + triangle[i - 1][j])
            row.append(1)
            triangle.append(row)

        return triangle
        
  
# https://leetcode.com/problems/pascals-triangle/
